lib/saml_idp/request.rb

```diff
@@ -121,74 +120,82 @@ def response_url
     end
 
     def log(msg)
-      if Rails && Rails.logger
+      if Rails&.logger
         Rails.logger.info msg
       else
         puts msg
       end
     end
 
     def valid?
-      log "Checking validity..."
+      log 'Checking validity...'
 
       unless service_provider?
         log "Unable to find service provider for issuer #{issuer}"
-        return false
+        @errors.push('Issuer is missing or invalid')
```

Comment:
1. I would consider using symbols for the errors in case we ever want to try translating these, or iterating on the message we want to show in the API, we can do that in the IDP directly

2. I'd use the accessor method instead of the ivar directly because it helps catch typos earlier  

so like:
 ```suggestion
         errors.push(:issuer_missing_or_invalid)
```

Reply:
Another idea! Make errors an `ActiveModel::Errors` instance so we can use the same `errors.add(....)` we use in the IDP

Reply:
oh yeah!!! this is exactly what i want to do. thanks!

Reply:
i incorporated the symbols/using accessor method in https://github.com/18F/saml_idp/pull/82/commits/872d7eeb20f10a698588aa47dfe9d2fa0118261f ... i don't think there is a way to use `ActiveModel::Errors` since there is no model involved :( so i left it as an array